sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/TestEnvironment.cs

```diff
@@ -35,24 +35,40 @@ public static class TestEnvironment
         private static readonly Lazy<string> EventHubsSecretInstance =
             new Lazy<string>(() => ReadAndVerifyEnvironmentVariable("EVENT_HUBS_SECRET"), LazyThreadSafetyMode.PublicationOnly);
 
+        /// <summary>The environment variable value for the namespace connection string, lazily evaluated.</summary>
+        private static readonly Lazy<string> EventHubsNamespaceConnectionString =
+            new Lazy<string>(() => ReadConnectionStringFromEnvironment("EVENT_HUBS_NAMESPACE_CONNECTION_STRING"), LazyThreadSafetyMode.PublicationOnly);
+
+        /// <summary>The environment variable value for the event hub name, lazily evaluated.</summary>
+        private static readonly Lazy<string> EventHubsEventHubName =
+            new Lazy<string>(() => ReadEnvironmentVariable("EVENT_HUBS_EVENT_HUB_NAME"), LazyThreadSafetyMode.PublicationOnly);
+
         /// <summary>The active Event Hubs namespace for this test run, lazily created.</summary>
         private static readonly Lazy<EventHubScope.NamespaceProperties> ActiveEventHubsNamespace =
-            new Lazy<EventHubScope.NamespaceProperties>(CreateNamespace, LazyThreadSafetyMode.ExecutionAndPublication);
+            new Lazy<EventHubScope.NamespaceProperties>(CreateNamespaceIfMissing, LazyThreadSafetyMode.ExecutionAndPublication);
 
         /// <summary>The fully qualified namespace contained within the active connection string, lazily created.</summary>
         private static readonly Lazy<string> FullyQualifiedNamespaceInstance =
             new Lazy<string>(() => ParseFullyQualifiedNamespace(EventHubsConnectionString), LazyThreadSafetyMode.PublicationOnly);
 
+        /// <summary>The shared access key name contained within the active connection string, lazily created.</summary>
+        private static readonly Lazy<string> SharedAccessKeyNameInstance =
+            new Lazy<string>(() => ParseSharedAccessKeyName(EventHubsConnectionString), LazyThreadSafetyMode.PublicationOnly);
+
+        /// <summary>The shared access key contained within the active connection string, lazily created.</summary>
+        private static readonly Lazy<string> SharedAccessKeyInstance =
+            new Lazy<string>(() => ParseSharedAccessKey(EventHubsConnectionString), LazyThreadSafetyMode.PublicationOnly);
+
         /// <summary>The name of the shared access key to be used for accessing an Event Hubs namespace.</summary>
         public const string EventHubsDefaultSharedAccessKey = "RootManageSharedAccessKey";
 
         /// <summary>
         ///   Indicates whether or not an ephemeral namespace was created for the current test execution.
         /// </summary>
         ///
-        /// <value><c>true</c> if an Event Hubs namespace was created; otherwise, <c>false</c>.</value>
+        /// <value><c>true</c> if an Event Hubs namespace was created from the current test run; otherwise, <c>false</c>.</value>
```

Comment:
(extreme) nit:  This reads a bit awkward to me.  Maybe something like `... was created for the current test run; ...`?